src/databricks/labs/remorph/transpiler/execute.py

```diff
@@ -215,11 +215,10 @@ def verify_workspace_client(workspace_client: WorkspaceClient) -> WorkspaceClien
 
     TODO: In future refactor this function so it can be used for reconcile module without cross access.
     """
-    product_info = workspace_client.config._product_info
-    if product_info[0] != "remorph":
-        product_info[0] = "remorph"
-    if product_info[1] != __version__:
-        product_info[1] = __version__
+    if workspace_client.config._product != "remorph":
+        workspace_client.config._product = "remorph"
+    if workspace_client.config._product_version != __version__:
+        workspace_client.config._product_version = __version__
     return workspace_client
```

Comment:
Assignment of the tuple back to workspce_client is missing

```suggestion
    workspace_client.config._product_info = product_info
    return workspace_client
```

Reply:
we're tapping directly into the `workspace_client.config._product_info` tuple so this is not necessary

Reply:
If we have different structures between 'externally created' then we need a better fix

Reply:
maybe you did not upgrade the sdk ?

Reply:
@ericvergnaud  refactored to use reflection instead